src/map/mob.cpp

```diff
@@ -1908,6 +1908,8 @@ static bool mob_ai_sub_hard(struct mob_data *md, t_tick tick)
 			md->state.skillstate = MSS_LOOT;
 			if (!unit_walktobl(&md->bl, tbl, 0, 0))
 				mob_unlocktarget(md, tick); //Can't loot...
+			else
+				md->ud.target = tbl->id; //Remember current loot target
```

Comment:
I want the change to influence as little as possible, so I only put it here, as it's only needed when unit_walktobl was successful. The only time we don't want to call unit_walktobl again is when the looter is already going for the loot.

I'll check the function. I think unit_set_target is for attack targets.

Reply:
Yeah I checked, we shouldn't use unit_set_target because the loot doesn't need to remember which monsters go after it. That function's purpose is to keep track of the number of attackers currently attacking you.

Reply:
Okay after debugging, I can use unit_set_target.

It will do nothing because: `&& (target = map_id2bl(target_id))` just returns false (nullptr is interpreted as false). So both variants would do the same, but I guess it might be cleaner to only have one place setting ud.target directly.